### DISCUSSION TITLE
Renovate suggested an older version for alpine 3.16 zip

### DISCUSSION THREAD

**Discussion by adamdavis40208** [renovatebot/renovate#24708] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

<img width="1639" alt="image" src="https://github.com/renovatebot/renovate/assets/19850646/62323588-3b13-4280-90c4-3324aa2b79d7">
For some reason, renovate keeps trying to downgrade our r12 to r9

It feels like a "r1 < r9" type issue

```
# renovate: datasource=repology depName=alpine_3_16/zip versioning=loose
ENV ZIP_VERSION="3.0-r12"
```

Minimal repo:
https://github.com/adamdavis40208/alpine-zip-renovate-issue/pull/2/files

Job that ran:
https://developer.mend.io/github/adamdavis40208/alpine-zip-renovate-issue/-/job/c3024c82-27f5-4d44-95f2-60458c73de57

### Relevant debug logs

```
          {
            "currentValue": "3.0-r12",
            "currentVersion": "3.0-r12",
            "datasource": "repology",
            "depName": "alpine_3_16/zip",
            "fixedVersion": "3.0-r12",
            "isSingleVersion": true,
            "packageName": "alpine_3_16/zip",
            "registryUrl": "https://repology.org",
            "replaceString": "# renovate: datasource=repology depName=alpine_3_16/zip versioning=loose\nENV ZIP_VERSION=\"3.0-r12\"\n",
            "versioning": "loose",
            "warnings": [],
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "3.0-r9",
                "newValue": "3.0-r9",
                "newMajor": 3,
                "newMinor": 0,
                "updateType": "patch",
                "branchName": "renovate-all-minor-patch"
              }
            ]
          },
```

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by msw-kialo** [comment, marked as answer]
Try out [`deb`](https://docs.renovatebot.com/modules/versioning/#deb-version-versioning) as versioning instead of `loose`. While `deb` is designed for apt/deb based Linux distributions, its sorting behavior is generic and a good fit for all/most distributions.

Specifically, `loose` falls back to a character sorting after a semver part (`3.0` in this case): `-r9` is greater than `-r12` that way. However, `deb` will always sort digits numerically (even if found in the suffix).

**Reply by adamdavis40208** [reply]
Welp. :TIL: 

This worked great in the sample repo, thanks for the help!
